rllib/core/rl_trainer/tests/test_trainer_runner.py

```diff
@@ -26,125 +87,114 @@ def tearDown(cls) -> None:
-        runner = TrainerRunner(
-            trainer_class, trainer_cfg, compute_config=dict(num_gpus=2)
-        )
-
-        reader = get_cartpole_dataset_reader(batch_size=500)
-
-        min_loss = float("inf")
-        for iter_i in range(1000):
-            batch = reader.next()
-            results_worker_0, results_worker_1 = runner.update(batch.as_multi_agent())
-
-            loss = (
-                results_worker_0["loss"]["total_loss"]
-                + results_worker_1["loss"]["total_loss"]
-            ) / 2
-            min_loss = min(loss, min_loss)
-            print(f"[iter = {iter_i}] Loss: {loss:.3f}, Min Loss: {min_loss:.3f}")
-            # The loss is initially around 0.69 (ln2). When it gets to around
-            # 0.57 the return of the policy gets to around 100.
-            if min_loss < 0.57:
-                break
-            self.assertEqual(
-                results_worker_0["mean_weight"]["default_policy"],
-                results_worker_1["mean_weight"]["default_policy"],
-            )
-        self.assertLess(min_loss, 0.57)
 
-    def test_add_remove_module(self):
-        env = gym.make("CartPole-v1")
-        trainer_class = BCTfRLTrainer
-        trainer_cfg = dict(
-            module_class=DiscreteBCTFModule,
-            module_kwargs={
-                "observation_space": env.observation_space,
-                "action_space": env.action_space,
-                "model_config": {"hidden_dim": 32},
-            },
-            optimizer_config={"lr": 1e-3},
-            in_test=True,
-        )
-        runner = TrainerRunner(
-            trainer_class, trainer_cfg, compute_config=dict(num_gpus=2)
-        )
-
-        reader = get_cartpole_dataset_reader(batch_size=500)
-        batch = reader.next()
-
-        # update once with the default policy
-        results = runner.update(batch.as_multi_agent())
-        module_ids_before_add = {DEFAULT_POLICY_ID}
-        new_module_id = "test_module"
-
-        # add a test_module
-        runner.add_module(
-            module_id=new_module_id,
-            module_cls=DiscreteBCTFModule,
-            module_kwargs={
-                "observation_space": env.observation_space,
-                "action_space": env.action_space,
-                "model_config": {"hidden_dim": 32},
-            },
-            optimizer_cls=tf.keras.optimizers.Adam,
-        )
-
-        # do training that includes the test_module
-        results = runner.update(
-            MultiAgentBatch(
-                {new_module_id: batch, DEFAULT_POLICY_ID: batch}, batch.count
-            )
-        )
-
-        # check that module weights are updated across workers and synchronized
-        for i in range(1, len(results)):
-            for module_id in results[i]["mean_weight"].keys():
-                assert (
-                    results[i]["mean_weight"][module_id]
-                    == results[i - 1]["mean_weight"][module_id]
+        for fw in ["tf", "torch"]:
```

Comment:
Also, shouldn't this be "tf2" everywhere from here on (new APIs don't support "tf" (static graph) anymore)?

Reply:
Cannot use framework iterator because this config here is just trainer_runner config and not algorithm_config. We can decide on the tf vs. tf2 name later. I think if we deprecate tf1 then there should be only tf. But for transitional period yes. This is a unittest and has nothing to do with the algorithm_config conventions.

Reply:
Got it! Makes sense.